### ISSUE TITLE
Cannot start judgedaemon as a service

### ISSUE THREAD

**Issue by ykazakov** [DOMjudge/domjudge#855]
### Description of the problem

When setting up a [judgehost](https://www.domjudge.org/docs/manual/7.3/install-judgehost.html#) under ubuntu 20.04 container, I cannot seem to enable the `create-cgroups` and `domjudge-judgehost` services:

```
$ sudo systemctl enable create-cgroups --now
Failed to enable unit: Unit file create-cgroups.service does not exist.
$ sudo systemctl enable domjudge-judgehost
Failed to enable unit: Unit file domjudge-judgehost.service does not exist.
```

It looks like these services are installed under `domjudge/lib/systemd/system/`, but I am not sure if `systemctl` finds them there.

If I manually create symlinks, the services can be enabled but  `domjudge-judgehost` still does not start:

```
$ sudo ln -s ~/domjudge/lib/systemd/system/* /etc/systemd/system/
$ sudo systemctl enable create-cgroups --now
Created symlink /etc/systemd/system/multi-user.target.wants/create-cgroups.service → /home/ubuntu/domjudge/lib/systemd/system/create-cgroups.service.
$ sudo systemctl enable domjudge-judgehost
Created symlink /etc/systemd/system/multi-user.target.wants/domjudge-judgehost.service → /home/ubuntu/domjudge/lib/systemd/system/domjudge-judgehost.service.
$ sudo systemctl start  domjudge-judgehost
$ systemctl status domjudge-judgehost
● domjudge-judgehost.service - DOMjudge JudgeDaemon
     Loaded: loaded (/home/ubuntu/domjudge/lib/systemd/system/domjudge-judgehost.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Mon 2020-07-27 20:13:31 UTC; 1s ago
    Process: 17369 ExecStart=/home/ubuntu/domjudge/judgehost/bin/judgedaemon -n 0 (code=exited, status=1/FAILURE)
   Main PID: 17369 (code=exited, status=1/FAILURE)
```

Looks like the service tries to run the judge under a user `domjudge-run-0` not `domjudge-run`. This is probably due to the `-n 0` option:

```
$ /home/ubuntu/domjudge/judgehost/bin/judgedaemon -n 0
[Jul 27 20:14:17.800] judgedaemon[17463]: error: runuser domjudge-run-0 does not exist.
```

Is it how it suppose to work, or am I missing something? 

### Your environment
- DOMjudge 7.3.0
- Ubuntu 20.04 server running inside an LXD container (not sure if that matters)
- LXC version 4.3
- $ uname -a
Linux judgehost2 5.4.0-42-generic #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

### Steps to reproduce
```
$ # create a fresh lxd container based on ubuntu 20.04 
$ lxc launch ubuntu:focal judgehost2
$ # make sure that mknod works since the container is unprivileged 
$ lxc config set judgehost2 security.syscalls.intercept.mknod true
$ lxc restart judgehost2
$ # enter the container under the user ubuntu
$ lxc exec judgehost2 -- su ubuntu -l
$ # download and unpack domjudge
$ wget https://www.domjudge.org/releases/domjudge-7.3.0.tar.gz
$ tar xzf domjudge-7.3.0.tar.gz
$ cd domjudge-7.3.0/
$ # install required packages
$ sudo apt update
$ sudo apt upgrade -y
$ sudo apt install -y make debootstrap libcgroup-dev php-cli php-curl php-json php-xml php-zip gcc g++ default-jre-headless default-jdk-headless ghc fp-compiler libcurl4-nss-dev libjsoncpp-dev
$ vi etc/restapi.secret # paste the credentials from etc/restapi.secret of DOMserver
$ # building and installing judgehost according to the manual
$ ./configure --prefix=$HOME/domjudge
$  make judgehost
$ sudo make install-judgehost
$ sudo useradd -d /nonexistent -U -M -s /bin/false domjudge-run
$  sudo cp etc/sudoers-domjudge /etc/sudoers.d/
$ # setup chroot
$ cd ../domjudge/judgehost/
$ sudo bin/dj_make_chroot -D Ubuntu -R bionic 
```

The CGroups options `cgroup_enable=memory swapaccount=1` are enabled in `/etc/default/grub` on the host.

### Any other information that you want to share?

If I remove the `-n 0` option from `domjudge-judgehost.service` then the judgehost starts and seem to connect fine to the domserver (as can be checked from the web interface). But I was not able to test it yet.

**Comment by eldering** [comment]
Indeed, when you use the `-n X` option with the `judgedaemon` you should create a user `domjudge-run-X`, since the processes of each instance should be separated from each other. This is indeed not clearly documented.

**Comment by ykazakov** [comment]
Can a judgehost run several jobs in parallel and all these jobs are run under different users? Is there a limit for a number of parallel jobs?
I am a bit puzzled, how the other users are chosen if the service runs just with `-n 0`? Or does it mean I can run several such services with different values of `-n` on the same machine and each will be visible as a separate judgehost?

**Comment by thijskh** [comment]
The first issue you encountered is probably #834 btw.

**Comment by ykazakov** [comment]
@thijskh I am not so sure. I cannot find these services under `/lib/systemd/system/`. Which script suppose to install these services?

**Comment by thijskh** [comment]
`make install` should do that

**Comment by ykazakov** [comment]
@thijskh 
```
$ make install
Try:
 - make install-domserver
 - make install-judgehost
 - make install-docs
make: *** [Makefile:37: install] Error 1
```

**Comment by eldering** [comment]
Well, `make install-judgehost` in this case.

**Comment by ykazakov** [comment]
I see only that they are only copied in the destination directory (`$HOME/domjudge` in my case):
```
install-judgehost:
        $(INSTALL_PROG) -t $(DESTDIR)$(judgehost_libjudgedir) \
                compile*.sh testcase_run.sh chroot-startstop.sh \
                check_diff.sh evict
        $(INSTALL_DATA) -t $(DESTDIR)$(judgehost_libjudgedir) \
                judgedaemon.main.php
        $(INSTALL_PROG) -t $(DESTDIR)$(judgehost_bindir) \
        judgedaemon runguard runpipe create_cgroups
        $(INSTALL_DIR)     $(DESTDIR)$(libdir)/systemd/system/
        $(INSTALL_DATA) -t $(DESTDIR)$(libdir)/systemd/system/ \
        create-cgroups.service domjudge-judgehost.service
```

**Comment by ykazakov** [comment]
I just tried to copy `domjudge-judgehost.service` with different values of the `-n` switch and start these services. They are indeed recognised as different judgehosts from the web interface! So I guess one can run multiple judge hosts off the same installation, which is quite useful if there are many CPUs available. Would be nice if this were documented.

```
$ systemctl status domjudge*
● domjudge-judgehost-1.service - DOMjudge JudgeDaemon
     Loaded: loaded (/usr/local/lib/systemd/system/domjudge-judgehost-1.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2020-07-28 19:41:59 UTC; 32min ago
   Main PID: 50788 (php)
      Tasks: 1 (limit: 4343)
     Memory: 26.3M
     CGroup: /system.slice/domjudge-judgehost-1.service
             └─50788 php /usr/local/judgehost/bin/judgedaemon -n 1

Jul 28 19:45:24 judgehost-vm2 sudo[66940]: pam_unix(sudo:session): session opened for user root by (uid=0)
Jul 28 19:45:24 judgehost-vm2 sudo[66940]: pam_unix(sudo:session): session closed for user root
Jul 28 19:45:24 judgehost-vm2 judgedaemon[50788]: [Jul 28 19:45:24.230] judgedaemon[50788]: Judging s198/j7088 finished
Jul 28 19:45:24 judgehost-vm2 judgedaemon[50788]: Judging s198/j7088 finished
Jul 28 19:45:24 judgehost-vm2 judgedaemon[50788]: 
Jul 28 19:45:24 judgehost-vm2 judgedaemon[50788]: API request POST judgehosts/next-judging/judgehost-vm2-1
Jul 28 19:45:24 judgehost-vm2 judgedaemon[50788]: 
Jul 28 19:45:24 judgehost-vm2 judgedaemon[50788]: [Jul 28 19:45:24.316] judgedaemon[50788]: No submissions in queue (for endpoint default), waiting...
Jul 28 19:45:24 judgehost-vm2 judgedaemon[50788]: No submissions in queue (for endpoint default), waiting...
Jul 28 19:45:24 judgehost-vm2 judgedaemon[50788]: 

● domjudge-judgehost.service - DOMjudge JudgeDaemon
     Loaded: loaded (/usr/local/lib/systemd/system/domjudge-judgehost.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2020-07-28 19:41:55 UTC; 32min ago
   Main PID: 50761 (php)
      Tasks: 1 (limit: 4343)
     Memory: 18.5M
     CGroup: /system.slice/domjudge-judgehost.service
             └─50761 php /usr/local/judgehost/bin/judgedaemon -n 0

Jul 28 19:45:44 judgehost-vm2 sudo[67044]: pam_unix(sudo:session): session opened for user root by (uid=0)
Jul 28 19:45:44 judgehost-vm2 sudo[67044]: pam_unix(sudo:session): session closed for user root
Jul 28 19:45:44 judgehost-vm2 judgedaemon[50761]: [Jul 28 19:45:44.918] judgedaemon[50761]: Judging s196/j7086 finished
Jul 28 19:45:44 judgehost-vm2 judgedaemon[50761]: Judging s196/j7086 finished
Jul 28 19:45:44 judgehost-vm2 judgedaemon[50761]: 
Jul 28 19:45:44 judgehost-vm2 judgedaemon[50761]: API request POST judgehosts/next-judging/judgehost-vm2-0
Jul 28 19:45:44 judgehost-vm2 judgedaemon[50761]: 
Jul 28 19:45:45 judgehost-vm2 judgedaemon[50761]: [Jul 28 19:45:45.001] judgedaemon[50761]: No submissions in queue (for endpoint default), waiting...
Jul 28 19:45:45 judgehost-vm2 judgedaemon[50761]: No submissions in queue (for endpoint default), waiting...
Jul 28 19:45:45 judgehost-vm2 judgedaemon[50761]:
```

**Comment by thijskh** [comment]
It’s documented here, isn’t it? https://www.domjudge.org/docs/manual/7.3/config-advanced.html#multiple-judgedaemons-per-machine

**Comment by ykazakov** [comment]
Thanks for the pointer @thijskh! I have overlooked this part (I was still struggling with the basic installation).
Nevertheless, it could be useful if `judgedaemon` also provides help about this option.

Here is my suggestion for the fixes so that the basic installation goes as smoothly as possible:

1. Fix the default prefix #856 
2. Remove the prefix `--prefix=$HOME/domjudge` from [the installation instructions](https://www.domjudge.org/docs/manual/7.3/install-judgehost.html#building-and-installing)
3. Remove the `-n 0` argument of `domjudge-judgehost.service`
4. Document the `-n` switch of the command `judgedaemon` command.
4. For the [advanced configuration](https://www.domjudge.org/docs/manual/7.3/config-advanced.html#multiple-judgedaemons-per-machine) perhaps make a hint how to start multiple judgedaemons as services (copy domjudge-judgehost.service many times or maybe there is a way to do that in one service?)
